7.GoogleCloudPlatform_MicroServices_Demo/README.md

```diff
@@ -0,0 +1,193 @@
+recommendationservice-v1-784888d964-9958l   1/1     Running   0          113s
+redis-cart-v1-797f4579d4-2xt4l              1/1     Running   0          113s
+shippingservice-v1-5bc9f577b7-kthts         1/1     Running   0          114s
+```
+
+### Access the web frontend in a browser
+
+The microservice **fronted** expose a a service(LoadBalacer) to allow outside access.
+
+```shell
+$ kubectl get service frontend-external
+NAME                TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
+frontend-external   LoadBalancer   <your ip>       <pending>     80:31420/TCP   66s
+```
+Although kind does not support type=LoadBalancer currently but we can still access the fronted from local.
+
+```shell
+kubectl port-forward svc/frontend 8000:80
+```
+
+Then you can access **Online Boutique** through [http://127.0.0.1:8000/](http://127.0.0.1:8000/)
+
+## Use istio for traffic management
+
+### Create istio gateway and virtualservice
+
+```
+kubectl apply -f istio-manifests.yaml
+```
+
+### Access the web from istio gateway
+
+```
+$ kubectl get svc/istio-ingressgateway -n istio-system
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                      AGE
+istio-ingressgateway   LoadBalancer   <your ip>       <pending>     15021:30077/TCP,80:31791/TCP,443:32113/TCP,31400:31457/TCP,15443:32035/TCP   12m
+```
+
+We do port forwarding to the ingressgateway of istio.
+
+```
+kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80
+```
+Then you can access the **Online Boutique** from [http://127.0.0.1:8080/](http://127.0.0.1:8080/).
+
+### Traffic Management
+In this section, we show the istio's traffic management capabilities. 
+
+1. Create an Istio DestinationRule for productcatalogservice.
+```
+kubectl apply -f istio-canary/destinationrule.yaml
+```
+
+2. Deploy ProductCatalog v2. We need open [online-boutique.yaml](./online-boutique.yaml)  and comment the Component productcatalogservice.
+
+``` YAML
+    # - name: productcatalogservice
+    #   type: microservice
+    #   settings:
+    #     image: gcr.io/google-samples/microservices-demo/productcatalogservice:v0.1.3
+    #     servicePort: 3550
+    #     env:
+    #       PORT: "3550"
+    #     cpu: "100m"
+    #     memory: "64Mi"
+    #   traits:
+    #     - name: patch-cmd-probe
+    #       properties:
+    #         readinessProbe:
+    #           cmd: ["/bin/grpc_health_probe", "-addr=:3550"]
+    #         livenessProbe:
+    #           cmd: ["/bin/grpc_health_probe", "-addr=:3550"]
+    #     - name: patch-container-resource
+    #       properties:
+    #         limitCPU: "200m"
+    #         limitMemory: "128Mi"
+```
+
+3. Uncomment the Component productcatalogservice-v2.
+
+``` YAML
+    - name: productcatalogservice
+      type: microservice
+      settings:
+        version: "v2"
+        image: gcr.io/google-samples/microservices-demo/productcatalogservice:v0.1.3
```

Comment:
please use a mirror image

Reply:
The image has been synchronized to Alibaba Cloud ACR